lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp

```diff
@@ -2673,36 +2673,45 @@ void mlir::torch::onnx_c::populateDefaultDomainGtoP(
 
         return success();
       });
-  patterns.onOp(
-      "LpNormalization", 1,
-      [](OpBinder binder, ConversionPatternRewriter &rewriter) {
-        Torch::ValueTensorType resultType;
-        int64_t axis, p;
-        Value input;
-        if (binder.tensorOperand(input) ||
-            binder.s64IntegerAttr(axis, "axis", -1) ||
-            binder.s64IntegerAttr(p, "p", 2) ||
-            binder.tensorResultType(resultType))
-          return failure();
-
-        auto loc = binder.getLoc();
-        Value cstAxis = rewriter.create<Torch::ConstantIntOp>(
-            loc, rewriter.getI64IntegerAttr(axis));
-        Value cstP = rewriter.create<Torch::ConstantIntOp>(
-            loc, rewriter.getI64IntegerAttr(p));
-        Value cstKeepDim = rewriter.create<Torch::ConstantBoolOp>(
-            loc, rewriter.getBoolAttr(true));
-        Value axisPrimList = rewriter.create<Torch::PrimListConstructOp>(
-            binder.getLoc(),
-            rewriter.getType<Torch::ListType>(
-                rewriter.getType<Torch::IntType>()),
-            llvm::ArrayRef<Value>{cstAxis});
-
-        rewriter.replaceOpWithNewOp<Torch::AtenNormScalarOptDimOp>(
-            binder.op, resultType, input, cstP, axisPrimList, cstKeepDim);
+  patterns.onOp("LpNormalization", 1,
+                [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+                  Torch::ValueTensorType resultType;
+                  int64_t axis, p;
+                  Value input;
+                  if (binder.tensorOperand(input) ||
```

Comment:
This is more a question for my benefit than yours.

I sometimes separate this into individual checks & give a different error message on each.

But also, this code looks cleaner.

Wonder what your thoughts on the tradeoffs are?

Reply:
I think the benefits to writing individual checks is minimal unless you have some reason to expect them to fail due to variation in versioning etc., but even then most of the binder methods don't actually fail since they fall back to a default value.